pkg/api/types.go

```diff
@@ -271,10 +271,10 @@ type KubernetesAddon struct {
 	Data       string                    `json:"data,omitempty"`
 }
 
-// IsEnabled returns if the addon is explicitly enabled, or the user-provided default if non explicitly enabled
-func (a *KubernetesAddon) IsEnabled(ifNil bool) bool {
```

Comment:
These methods have been changed to be a strict convenience "is this addon enabled?" and not overloaded to include "default override" functionality. This is to more clearly isolate the source of defaults enforcement.